app/javascript/app/service-provider-form.js

```diff
@@ -1,52 +1,68 @@
-$(function () {
-  if (!$('.service-provider-form').length) {
-    return;
-  }
-
+function ialOptionSetup() {
   // Selectors
-  const idProtocol = $('input[name="service_provider[identity_protocol]"]');
-  const ialLevel = $('#service_provider_ial');
-  const samlFields = $('.saml-fields');
-  const oidcFields = $('.oidc-fields');
-  const ialAttributesCheckboxes = $('.ial-attr-wrapper');
-  const fileContainer = $('#certificate-container');
-  const logoInput = $('.input-file');
-  const pemInputMessage = $('.js-pem-input-error-message');
-  const pemInput = $('.js-pem-input');
-  const redirectURI = $('#add-redirect-uri-field');
-  const failureToProofURL = $('.service_provider_failure_to_proof_url');
-  const returnToSpUrl = document.querySelector('#service_provider_return_to_sp_url');
-
+  const ialLevel = document.querySelector('#service_provider_ial');
```

Comment:
If querying by ID, `getElementById` is a bit more targeted, equally concise, and marginally more performant.

```suggestion
  const ialLevel = document.getElementById('service_provider_ial');
```